### PR TITLE
New hybrid sample app doing smartsync with notes

### PR DESCRIPTION
**Pull request by wmathurin**
For now pulling to a new branch called notesync - we can decide later what to do with it

### CODE REVIEW

File: samples/notesync/NoteSync.html [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+
+    render: function(eventName) {
+        $(this.el).html(this.template());
+        $(".search-key", this.el).val(this.model.getCriteria());
+        this.listView.setElement($("ul", this.el)).render();
+        return this;
+    },
+
+    syncDown: function() {
+        var target = {type:"custom", androidImpl:"com.salesforce.samples.notesync.ContentSoqlSyncTarget", query:"SELECT " + this.fieldlist.join(",") + " FROM ContentNote LIMIT 1000"};
```
Comment (wmathurin) [inline samples/notesync/NoteSync.html]: Notice the custom sync target with the androidImpl attribute to specify the implementation (I'll add a iosImpl once I have support for custom sync target on iOS).
I'll send a pull request later for the Android repo with the ContentSoqlSyncTarget class.

File: samples/notesync/NoteSync.html [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+
+    template: _.template($("#note-page").html()),
+
+    events: {
+        "change": "change",
+        "click .save": "save"
+    },
+
+    render: function(eventName) {
+        var templateData = _.extend({ContentDecoded: atob(this.model.get("Content"))}, this.model.toJSON());
```
Comment (wmathurin) [inline samples/notesync/NoteSync.html]: Decoding base64 note content for the view

Comment (wmathurin) [inline samples/notesync/NoteSync.html]: Moving the decoding deeper (e.g. at the time of sync down/up) would have advantages (less space used, and the field could actually be searchable) but would require custom handling during sync down (which we can now do through custom sync target) and during sync up (which we don't have at this point).

File: samples/notesync/bootconfig.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+    "remoteAccessConsumerKey": "3MVG9Iu66FKeHhIMmUBkEV8o6PfICh5xUnN89HyifhHcvSSBM3WUYyG9buZBezdrhdYilqlXHHbqamC5MqaKP", // mobile2
```
Comment (wmathurin) [inline samples/notesync/bootconfig.json]: We should take that out (the mobile2.t client id) when we released that sample app.

**Comment by bhariharan** [comment]
:+1:

**Pull request merged**